text/0000-unkind-static-libraries.md

```diff
@@ -0,0 +1,54 @@
+- Feature Name: better_static_kind
+- Start Date: 2016-02-05
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Adds a new `kind=better_static` (name entirely open to bikeshedding) that is used to link static libraries by passing them to the linker, unlike `kind=dynamic` which is intended for dynamic libraries and `kind=static` which has rustc bundle it instead of passing it to the linker.
+
+# Motivation
+[motivation]: #motivation
+
+There is currently no way to properly link a static library by passing it to the linker instead of bundling it with rustc. `kind=dynamic` doesn't work because it thinks the library is a dynamic library resulting in issues such as passing the library on to later linker invocations beyond the first immediate linker invocation and also on Windows, once dllimport is actually supported correctly, `kind=dynamic` would cause dllimport to be emitted for all the symbols which is incorrect for static libraries. `kind=static` doesn't work because instead of passing the library to the linker, rustc bundles it, which results in rustc not looking in the standard library paths for the library.
```

Comment:
I am very sympathetic to the aims of this RFC -- and I think I am persuaded that there is something missing in the current model, though I have to try to bring the details back into cache -- but I'd appreciate it you could edit it to maintain a more objective tone. I find phrases like "no way to properly link" or describing things as "weird" distracting, since they imply value judgements, which complicates the issue.

I'd rather see something like:

"The current set of linker options do not allow static linking to be handled by the linker. The current `link=static` causes rustc to bundle up the executables and `kind=dynamic` uses dynamic linking. However, on windows, a common configuration is to do blah blah blah."